File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        values = {'I':1, 'V':5, 'X': 10, 'L':50, 'C':100, 'D':500, 'M':1000}
        output = 0
        for i in range(len(s)):
            if i < (len(s)-1) and values[s[i]]<values[s[i+1]]:
                output -= values[s[i]]
            else:
                output += values[s[i]]
        return output
        

File: 0013-roman-to-integer/0013-roman-to-integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        map<char, int>mp;
        mp['I'] = 1;
        mp['V'] = 5;
        mp['X'] = 10;
        mp['L'] = 50;
        mp['C'] = 100;
        mp['D'] = 500;
        mp['M'] = 1000;

        int output = 0;
        int n = s.size();

        for (int i = 0; i<n; i++){
            if (i < n-1 && mp[s[i]]<mp[s[i+1]]){
                output -= mp[s[i]];
            }
            else (output += mp[s[i]]);

        }return output;
        
    }
};